src/cocotb_tools/runner.py

```diff
@@ -1234,8 +1277,8 @@ def _test_command(self) -> List[_Command]:
 class Verilator(Runner):
     """Implementation of :class:`Runner` for Verilator.
 
+    * ``waves`` argument *must* be given to :meth:`.build` if used and if ``gui`` is provided to :meth:`.test`.
```

Comment:
This I am a bit confused about. For Verilator, `waves` is used both in `build` and in `test`. For all other simulators, it is only used in one of the methods. 

Was the first part of this sentence missing earlier (i.e., one must provide `waves` to `build` if it is used in `test`), or can I remove this sentence?

(Not a Verilator user, but not obvious why one specify it in `build` if not required? Performance?)

Reply:
Verilator is the only simulator where waves support is required during building. Passing `waves=True` to the build step builds the binary with tracing support which takes considerably longer to build and produces considerably larger binaries. Then deciding to run the test with waves has a considerable effect on runtime performance, so we wanted the `waves` switch in `test` to be able to turn it on/off.